HandyHansel/Commands/PrefixCommands.cs

```diff
@@ -117,39 +129,51 @@ public async Task InteractiveRemovePrefix(CommandContext context)
+                await Task.Delay(5000);
+                await context.Channel.DeleteMessagesAsync(new List<DiscordMessage> { msg, snark });
                 return;
             }
 
-            await context.RespondAsync("Ok, which event do you want to remove?");
-
-            _ = interactivity.SendPaginatedMessageAsync(context.Channel, context.User,
-                GetGuildPrefixPages(interactivity, guildPrefixes),
-                behaviour: PaginationBehaviour.WrapAround, timeoutoverride: TimeSpan.FromMinutes(1));
+            DiscordEmbedBuilder removeEventEmbed = new DiscordEmbedBuilder()
+                .WithTitle("Select a prefix to remove by typing: <prefix number>")
+                .WithColor(context.Member.Color);
 
-            await context.RespondAsync("Choose a prefix by typing: <prefix number>");
-
-            InteractivityResult<DiscordMessage> result = await interactivity.WaitForMessageAsync(
-                xm => int.TryParse(xm.Content, out _) && xm.Author.Equals(context.User),
-                TimeSpan.FromMinutes(1));
-
-            if (result.TimedOut)
+            Task<(bool, int)> messageValidationAndReturn(MessageCreateEventArgs messageE)
             {
+                if (messageE.Author.Equals(context.User) && int.TryParse(messageE.Message.Content, out int eventToChoose))
+                {
+                    return Task.FromResult((true, eventToChoose));
+                }
+                else
+                {
+                    return Task.FromResult((false, -1));
+                }
+            }
+            await msg.DeleteAllReactionsAsync();
+            CustomResult<int> result = await interactivity.WaitForMessagePaginationOnMsg(
+                context,
+                this.GetGuildPrefixPages(guildPrefixes, interactivity, removeEventEmbed),
+                messageValidationAndReturn,
+                msg: msg);
+
+            if (result.TimedOut || result.Cancelled)
+            {
+                DiscordMessage snark = await context.RespondAsync("You never gave me a valid input. Thanks for wasting my time. :triumph:");
+                await Task.Delay(5000);
+                await context.Channel.DeleteMessagesAsync(new List<DiscordMessage> { msg, snark });
                 return;
             }
 
-            GuildPrefix selectedPrefix =
-                dataAccessProvider.GetAllAssociatedGuildPrefixes(context.Guild.Id)
-                    .ToList()[int.Parse(result.Result.Content) - 1];
+            GuildPrefix selectedPrefix = guildPrefixes[result.Result - 1];
 
-            dataAccessProvider.DeleteGuildPrefix(selectedPrefix);
+            provider.DeleteGuildPrefix(selectedPrefix);
 
-            await context.RespondAsync(
-                $"You have deleted the prefix \"{selectedPrefix.Prefix}\" from this guild's prefixes.");
+            await msg.ModifyAsync(
+                $"You have deleted the prefix \"{selectedPrefix.Prefix}\" from this guild's prefixes.", embed: null);
```

Comment:
Switch to using the specially defined extension method to respond to both messages and reactions